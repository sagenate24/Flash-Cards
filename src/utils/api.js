import { AsyncStorage } from 'react-native';
import {
  formatDeckResults,
  formatNewDeck,
  STORAGE_KEY,
  PROFILE_KEY,
  formatProfileResults,
} from './data';

// deck APIs.
export const getDecks = async () => {
  try {
    const decks = formatDeckResults(await AsyncStorage.getItem(STORAGE_KEY));

    return decks;
  } catch (error) {
    console.log(error.message);
  }
};

export const getDeck = async (deckTitle) => {
  try {
    const decks = JSON.parse(await AsyncStorage.getItem(STORAGE_KEY));

    return decks[deckTitle];
  } catch (error) {
    console.log(error.message);
  }
};

export const addCardToDeck = async (card, deckTitle) => {
  try {
    const data = JSON.parse(await AsyncStorage.getItem(STORAGE_KEY));
    const questionsArr = data[deckTitle].questions;
    questionsArr.push(card);

    await AsyncStorage.mergeItem(STORAGE_KEY, JSON.stringify({
      [deckTitle]: {
        questions: questionsArr,
      },
    }));
  } catch (error) {
    console.log('EERRRROORR OH MY GURD', error.message);
  }
};

export const addDeckTitle = async (title) => {
  try {
    await AsyncStorage.mergeItem(STORAGE_KEY, formatNewDeck(title));
  } catch (error) {
    console.log(error.message);
  }
};

export const removeDeck = async (key) => {
  try {
    const data = JSON.parse(await AsyncStorage.getItem(STORAGE_KEY));
    data[key] = undefined;
    delete data[key];

    await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(data));
  } catch (error) {
    console.log(error.message);
  }
};

export const recentActivityScore = async (deckTitle, score, timeStamp) => {
  try {
    await AsyncStorage.mergeItem(STORAGE_KEY, JSON.stringify({
      [deckTitle]: {
        recentScore: score,
        timeStamp,
      },
    }));
  } catch (error) {
    console.log(error.message);
  }
};

// profile APIs.
export const getProfile = async () => {
  try {
    const profile = await AsyncStorage.getItem(PROFILE_KEY)
      .then(results => formatProfileResults(results));

    return profile;
  } catch (error) {
    console.log(error.message);
  }
};

export const deleteProfile = async () => {
  try {
    const profile = await AsyncStorage.getItem(PROFILE_KEY)
      .then(() => formatProfileResults(null));

    return profile;
  } catch (error) {
    console.log(error.message);
  }
};

export const addProfileImg = async (image) => {
  try {
    await AsyncStorage.mergeItem(PROFILE_KEY, JSON.stringify({
      avatar: image,
    }));
  } catch (error) {
    console.log(error.message);
  }
};

export const addProfileCover = async (image) => {
  try {
    await AsyncStorage.mergeItem(PROFILE_KEY, JSON.stringify({
      cover: image,
    }));
  } catch (error) {
    console.log(error.message);
  }
};

export const addProfileName = async (username) => {
  try {
    await AsyncStorage.mergeItem(PROFILE_KEY, JSON.stringify({
      username,
    }));
  } catch (error) {
    console.log(error.message);
  }
};
