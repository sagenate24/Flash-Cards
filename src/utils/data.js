import { AsyncStorage } from 'react-native';

export const STORAGE_KEY = 'FlashCards:decks';
export const PROFILE_KEY = 'FlashCards:profile';

function getProfile(profile) {
  const info = {
    profile: {
      username: '',
      avatar: '',
      cover: '',
      parentControl: 'off',
    },
  };

  return typeof profile === 'undefined'
    ? info
    : info[profile];
}

function setDummyProfile() {
  const { profile } = getProfile();

  AsyncStorage.setItem(PROFILE_KEY, JSON.stringify(profile));

  return profile;
}

export function formatProfileResults(results) {
  return results === null
    ? setDummyProfile()
    : JSON.parse(results);
}

export function formatDeckResults(results) {
  return results === null
    ? null
    : JSON.parse(results);
}

export function formatNewDeck(deckTitle) {
  const timestamp = Date.now();

  return JSON.stringify({
    [deckTitle]: {
      title: deckTitle,
      timeStamp: timestamp,
      questions: [],
    },
  });
}
