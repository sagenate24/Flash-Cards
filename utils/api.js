import { AsyncStorage } from 'react-native';
import {
  formatDeckResults,
  formatNewDeck,
  STORAGE_KEY,
  PROFILE_KEY,
  formatProfileResults,
  formatNewProfile,
} from './_data';

// Deck APIs.
export const getDecks = async () => {
  try {
    const decks = await AsyncStorage.getItem(STORAGE_KEY)
      .then(results => formatDeckResults(results));

    return decks;
  } catch (error) {
    console.log(error.message);
  }
};

export const getDeck = async (deckTitle) => {
  try {
    const decks = await AsyncStorage.getItem(STORAGE_KEY)
      .then(results => JSON.parse(results));

    return decks[deckTitle];
  } catch (error) {
    console.log(error.message);
  }
};

export function addCardToDeck(card, deckTitle) {
  AsyncStorage.getItem(STORAGE_KEY, (err, data) => {
    const newData = JSON.parse(data);

    const questionsArr = newData[deckTitle].questions;
    questionsArr.push(card);

    AsyncStorage.mergeItem(STORAGE_KEY, JSON.stringify({
      [deckTitle]: {
        questions: questionsArr,
      },
    }), () => {
      AsyncStorage.getItem(STORAGE_KEY).then(results => console.log(JSON.parse(results)));
    });
  });
}

export const addDeckTitle = async (title) => {
  try {
    const newDeck = await AsyncStorage.mergeItem(STORAGE_KEY, formatNewDeck(title), () => {
      AsyncStorage.getItem(STORAGE_KEY);
    });

    return JSON.parse(newDeck);
  } catch (error) {
    console.log(error.message);
  }
};

export function removeDeck(key) {
  return AsyncStorage.getItem(STORAGE_KEY)
    .then((results) => {
      const data = JSON.parse(results);
      data[key] = undefined;
      delete data[key];
      AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(data)), () => {
        AsyncStorage.getItem(PROFILE_KEY).then(results => console.log(JSON.parse(results)));
      };
    });
}

export function recentActivityScore(deckTitle, score, timeStamp) {
  AsyncStorage.getItem(STORAGE_KEY, () => {
    AsyncStorage.mergeItem(STORAGE_KEY, JSON.stringify({
      [deckTitle]: {
        recentScore: score,
        timeStamp,
      },
    }), () => {
      AsyncStorage.getItem(STORAGE_KEY).then(results => console.log(JSON.parse(results)));
    });
  });
}

// Profile APIs.
export const getProfile = async () => {
  try {
    const profile = await AsyncStorage.getItem(PROFILE_KEY)
      .then(results => formatProfileResults(results));

    return profile;
  } catch (error) {
    console.log(error.message);
  }
};

export const deleteProfile = async () => {
  try {
    const profile = await AsyncStorage.getItem(PROFILE_KEY)
      .then(() => formatNewProfile());

    return profile;
  } catch (error) {
    console.log(error.message);
  }
};

export function addProfileImg(image) {
  AsyncStorage.getItem(PROFILE_KEY, () => {
    AsyncStorage.mergeItem(PROFILE_KEY, JSON.stringify({
      avatar: image,
    }), () => {
      AsyncStorage.getItem(PROFILE_KEY).then(results => console.log(JSON.parse(results)));
    });
  });
}

export function addProfileCover(image) {
  AsyncStorage.getItem(PROFILE_KEY, () => {
    AsyncStorage.mergeItem(PROFILE_KEY, JSON.stringify({
      cover: image,
    }), () => {
      AsyncStorage.getItem(PROFILE_KEY).then(results => console.log(JSON.parse(results)));
    });
  });
}

export function addProfileName(username) {
  AsyncStorage.getItem(PROFILE_KEY, () => {
    AsyncStorage.mergeItem(PROFILE_KEY, JSON.stringify({
      username,
    }), () => {
      AsyncStorage.getItem(PROFILE_KEY).then(results => console.log(JSON.parse(results)));
    });
  });
}
