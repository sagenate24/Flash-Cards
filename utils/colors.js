export const gray = '#e1e8f0';
export const lightGray = '#f1f3f4';
export const white = '#fff';
export const black = '#404040';
export const red = '#e63946';
export const lightBlue = '#a8dadc';
export const queenBlue = '#457b9d';
export const yellow = '#e3b505';
