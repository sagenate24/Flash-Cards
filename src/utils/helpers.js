import { Notifications, Permissions } from 'expo';
import { AsyncStorage } from 'react-native';
import Filter from 'bad-words';
import { getDecks, getProfile } from './api';

const filter = new Filter();

const NOTIFICATION_KEY = 'FlashCards:notifications';

export function profanityDetector(text) {
  return filter.clean(text);
}

export const getInitialData = () => Promise.all([
  getDecks(),
  getProfile(),
]).then(([decks, profile]) => ({
  decks,
  profile,
}));

export function timeToString(timestamp) {
  const date = new Date(timestamp);
  const todayUTC = new Date(date.getFullYear(), date.getMonth(), date.getDate());
  return todayUTC.toDateString().substring(4);
}

export function clearLocalNotification() {
  return AsyncStorage.removeItem(NOTIFICATION_KEY)
    .then(Notifications.cancelAllScheduledNotificationsAsync);
}

export function createNotification() {
  return {
    title: 'Time to study!',
    body: "ğŸ‘‹ Don't forget to quiz yourself today!",
    ios: {
      sound: true,
    },
    android: {
      sounds: true,
      priority: 'high',
      sticky: false,
      vibrate: true,
    },
  };
}

export function setLocalNotification() {
  AsyncStorage.getItem(NOTIFICATION_KEY)
    .then(JSON.parse)
    .then((data) => {
      if (data === null) {
        Permissions.askAsync(Permissions.NOTIFICATIONS)
          .then(({ status }) => {
            if (status === 'granted') {
              Notifications.cancelAllScheduledNotificationsAsync();

              const tomorrow = new Date();
              tomorrow.setDate(tomorrow.getDate() + 1);
              tomorrow.setHours(20);
              tomorrow.setMinutes(0);

              Notifications.scheduleLocalNotificationAsync(
                createNotification(),
                {
                  time: tomorrow,
                  repeat: 'day',
                },
              );

              AsyncStorage.setItem(NOTIFICATION_KEY, JSON.stringify(true));
            }
          });
      }
    });
}

export const askPermissionsAsync = async () => {
  const result = await Permissions.askAsync(Permissions.CAMERA_ROLL)
    .then(({ status }) => status);

  return result;
};
