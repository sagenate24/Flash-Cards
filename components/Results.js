import React, { Component } from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { addScore } from '../actions/decks';
import { recentActivityScore } from '../utils/api';
import { clearLocalNotification, setLocalNotification } from '../utils/helpers';
import { white, black, red, queenBlue } from '../utils/colors';

import CircleScore from './CircleScore';
import NASBtn from './NASBtn';

// buttons in andriod need to be fixed.

class Results extends Component {
  componentDidMount() {
    const { deck, correctAnswers, dispatch } = this.props;
    const correctPercent = Math.round((correctAnswers / deck.questions.length) * 100);
    const timeStamp = Date.now();

    recentActivityScore(deck.title, correctPercent, timeStamp);
    dispatch(addScore(deck.title, correctPercent, timeStamp));

    clearLocalNotification().then(() => {
      setLocalNotification();
    });
  }

  handlePercent = () => {
    const { deck, correctAnswers } = this.props;
    const correctPercent = Math.round((correctAnswers / deck.questions.length) * 100);

    return correctPercent;
  }

  goTo = (view) => {
    const { navigation, deck } = this.props;

    navigation.navigate(
      view,
      { currentDeck: deck },
    );
  }

  render() {
    const { deck, correctAnswers } = this.props;
    const percent = this.handlePercent();

    return (
      <View style={styles.container}>
        <View style={styles.card}>
          <Text style={styles.header}>Your Score</Text>
          <CircleScore
            percent={percent}
            width={5}
            size={130}
            textSize={{ fontSize: 34 }}
          />
          <View>
            {percent >= 80
              ? <Text style={styles.reviewText}>Great Job!</Text>
              : <Text style={styles.reviewText}>Study Time</Text>
            }
          </View>
          <View>
            <Text>You got {correctAnswers} out of {deck.questions.length} correct.</Text>
          </View>
          <NASBtn
            tintColor={[styles.button, { backgroundColor: red }]}
            onPress={() => this.goTo('Quiz')}
          >
            Retake
          </NASBtn>
          <NASBtn
            tintColor={[styles.button, { backgroundColor: queenBlue }]}
            onPress={() => this.goTo('Deck')}
          >
            Study more
          </NASBtn>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: white,
    flex: 1,
    alignItems: 'stretch',
    paddingLeft: 20,
    paddingRight: 20,
  },
  card: {
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 50,
    paddingBottom: 50,
    alignItems: 'center',
  },
  header: {
    paddingBottom: 30,
    fontSize: 26,
    color: black,
    fontWeight: 'bold',
    justifyContent: 'flex-start',
  },
  reviewText: {
    fontSize: 34,
    fontWeight: 'bold',
    color: black,
  },
  button: {
    width: 200,
    padding: 10,
    marginTop: 20,
  },
});

function mapStateToProps(state, { navigation }) {
  const { correctAnswers, currentDeck } = navigation.state.params;

  return {
    correctAnswers,
    deck: currentDeck,
  };
}

export default connect(mapStateToProps)(Results);
