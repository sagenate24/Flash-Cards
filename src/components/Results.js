import React, { Component } from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { addScore } from '../actions/decks';
import { recentActivityScore } from '../utils/api';
import { white, black, red, queenBlue } from '../utils/colors';
import { clearLocalNotification, setLocalNotification } from '../utils/helpers';

import NASBtn from './NASBtn';
import CircleScore from './CircleScore';

class Results extends Component {
  componentDidMount() {
    const { deck, correctAnswers, dispatch } = this.props;
    const correctPercent = Math.round((correctAnswers / deck.questions.length) * 100);
    const timeStamp = Date.now();

    recentActivityScore(deck.title, correctPercent, timeStamp);
    dispatch(addScore(deck.title, correctPercent, timeStamp));

    clearLocalNotification().then(() => {
      setLocalNotification();
    });
  }

  handlePercent = () => {
    const { deck, correctAnswers } = this.props;
    const correctPercent = Math.round((correctAnswers / deck.questions.length) * 100);

    return correctPercent;
  }

  goTo = (view) => {
    const { navigation, deck } = this.props;

    navigation.navigate(
      view,
      { currentDeck: deck },
    );
  }

  render() {
    const { deck, correctAnswers } = this.props;
    const percent = this.handlePercent();

    return (
      <View style={styles.container}>
        <View style={styles.card}>
          <Text style={styles.header}>Your Score</Text>
          <CircleScore
            width={5}
            size={130}
            percent={percent}
            textSize={{ fontSize: 34 }}
          />
          <View>
            {percent >= 80
              ? <Text style={styles.reviewText}>Great Job!</Text>
              : <Text style={styles.reviewText}>Study Time</Text>
            }
          </View>
          <View>
            <Text>You got {correctAnswers} out of {deck.questions.length} correct.</Text>
          </View>
          <NASBtn
            onPress={() => this.goTo('Quiz')}
            textStyle={{ color: red }}
            tintColor={[styles.button, { borderColor: red, borderWidth: 3, backgroundColor: white }]}
          >
            Retake
          </NASBtn>
          <NASBtn
            onPress={() => this.goTo('Deck')}
            textStyle={{ color: queenBlue }}
            tintColor={[styles.button, { borderColor: queenBlue, borderWidth: 3, backgroundColor: white }]}
          >
            Study more
          </NASBtn>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: white,
    flex: 1,
    alignItems: 'stretch',
    paddingLeft: 20,
    paddingRight: 20,
  },
  card: {
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 50,
    paddingBottom: 50,
    alignItems: 'center',
  },
  header: {
    paddingBottom: 30,
    fontSize: 26,
    color: black,
    fontWeight: 'bold',
    justifyContent: 'flex-start',
  },
  reviewText: {
    fontSize: 34,
    fontWeight: 'bold',
    color: black,
  },
  button: {
    width: 200,
    padding: 10,
    marginTop: 20,
  },
});

function mapStateToProps(state, { navigation }) {
  const { correctAnswers, currentDeck } = navigation.state.params;

  return {
    correctAnswers,
    deck: currentDeck,
  };
}

export default connect(mapStateToProps)(Results);
