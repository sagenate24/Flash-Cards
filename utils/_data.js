import { AsyncStorage } from 'react-native'
import { getDeckMetaInfo, timeToString } from './helpers'

export const STORAGE_KEY = 'FlashCards:decks'

function setDummyData () {
  const { React, JavaScript } = getDeckMetaInfo()

  let dummyData = {}
  const timestamp = Date.now()
  dummyData = {
    React,
    JavaScript,
  }

  AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(dummyData))

  return dummyData
}

export function formatDeckResults () {
  return setDummyData()
}

export function formatNewDeck(deckTitle) {
  const timestamp = Date.now();

  return JSON.stringify({
    [deckTitle]: {
      title: deckTitle,
      timestamp: timestamp,
      questions: [],
    },
  })
}