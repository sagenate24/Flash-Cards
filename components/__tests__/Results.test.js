import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { recentActivityScore } from '../../utils/api';

import Results from '../Results';

jest.mock('../../utils/api');

const mockCurrentDeck = {
  title: 'React',
  timeStamp: 1534284894237,
  recentScore: 35,
  questions: [
    {
      question: 'What is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is ReactWhat is React?',
      answer: 'A library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfacesA library for managing user interfaces',
    },
    {
      question: 'Where do you make Ajax requests in React?',
      answer: 'The componentDidMount lifecycle event',
    },
  ],
};

it('should flip card', (done) => {
  const wrapper = shallow(<Results.WrappedComponent deck={mockCurrentDeck} correctAnswers={1} dispatch={jest.fn()} />).instance();
  const handlePercent = wrapper.handlePercent();

  expect(handlePercent).toEqual(50);

  done();
});

it('should navigate correctly', () => {
  const mockNavigation = {
    navigate: jest.fn(),
  };

  const wrapper = shallow(<Results.WrappedComponent deck={mockCurrentDeck} correctAnswers={1} dispatch={jest.fn()} navigation={mockNavigation} />).instance();
  wrapper.goTo('Deck');

  expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
});

it('should save the score from the quiz', () => {
  const mockDispatch = jest.fn();
  const wrapper = shallow(<Results.WrappedComponent deck={mockCurrentDeck} correctAnswers={1} dispatch={mockDispatch} />).instance();
  const nowTime = Date.now();
  wrapper.componentDidMount();

  expect(recentActivityScore).toHaveBeenCalled();
  expect(recentActivityScore).toHaveBeenCalledWith('React', 50, nowTime);
  expect(mockDispatch).toHaveBeenCalled();
});
