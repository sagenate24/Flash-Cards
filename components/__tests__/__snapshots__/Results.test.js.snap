// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingBottom": 50,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#404040",
          "fontSize": 26,
          "fontWeight": "bold",
          "justifyContent": "flex-start",
          "paddingBottom": 30,
        }
      }
    >
      Your Score
    </Text>
    <View
      style={undefined}
    >
      <RNSVGSvgView
        bbHeight="130"
        bbWidth="130"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "backgroundColor": "transparent",
            },
            false,
            Object {
              "flex": 0,
              "height": 130,
              "width": 130,
            },
          ]
        }
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              0,
              0,
              0,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              -0.3826834323650895,
              -0.9238795325112868,
              0.9238795325112868,
              -0.3826834323650895,
              29.822253490497175,
              149.92659271696448,
            ]
          }
          onLayout={undefined}
          opacity={1}
          originX={65}
          originY={65}
          propList={Array []}
          rotation={247.5}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth="1"
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M 20.823184594520406 109.21152542754186 A 62.5 62.5 0 1 0 65 2.5"
            fill={
              Array [
                0,
                0,
                0,
                0,
                0,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            onLayout={undefined}
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={
              Array [
                0,
                0.9490196078431372,
                0.9490196078431372,
                0.9490196078431372,
                1,
              ]
            }
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={1}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth="5"
            x={0}
            y={0}
          />
          <RNSVGPath
            d="M 65 2.5 A 62.5 62.5 0 0 0 65 2.5"
            fill={
              Array [
                0,
                0,
                0,
                0,
                0,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            onLayout={undefined}
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={
              Array [
                0,
                0.8901960784313725,
                0.7098039215686275,
                0.0196078431372549,
                1,
              ]
            }
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={1}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth="5"
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 60,
            "height": 120,
            "justifyContent": "center",
            "left": 5,
            "position": "absolute",
            "top": 5,
            "width": 120,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 34,
              },
              Object {
                "color": "#404040",
                "fontWeight": "bold",
              },
            ]
          }
        >
          50
          %
        </Text>
      </View>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#404040",
            "fontSize": 34,
            "fontWeight": "bold",
          }
        }
      >
        Study Time
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        You got 
        1
         out of 
        2
         correct.
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#e63946",
            "borderRadius": 0,
            "marginTop": 20,
            "opacity": 1,
            "padding": 10,
            "width": 200,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 22,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Retake
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#457b9d",
            "borderRadius": 0,
            "marginTop": 20,
            "opacity": 1,
            "padding": 10,
            "width": 200,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 22,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Study more
        </Text>
      </View>
    </View>
  </View>
</View>
`;
